mixins/d2l-sequences-completion-tracking-mixin.js

```
@@ -18,36 +18,51 @@ function CompletionTrackingMixin() {
 				_completionEntity: {
 					type: Object
 				},
-				_failedCompletion: {
-					type: Object
-				},
 				_skipCompletion: {
 					type: Boolean,
 					computed: '_isImpersonating(token)'
 				}
 			};
 		}
 
-		finishCompletion() {
-			this._fireToastEvent();
-
-			if (!this._completionEntity || this._skipCompletion) {
+		startCompletion() {
+			if (!this.entity || this._skipCompletion || this._isComplete(this.entity)) {
 				return;
 			}
 
-			this._performViewActions(this._completionEntity, 'finish-view-activity')
-				.then(() => { this._completionEntity = null; })
-				.catch(() => { this._completionEntity = null; });
+			this._performViewActions(this.entity, 'view-activity-duration')
+				.then(completion => {
+					this._completionEntity = completion;
+					this._finishCompletion();
+				})
+				.catch(() => {});
```

Catching and silencing errors doesn't sit right with me. This is how we can get things dying without any errors in the console.

---

Agreed, not a good approach. Should I just `console.error` the error? Not sure how errors are typically handled, or **should** be handled. I see a lot of silenced errors here and there but I'm assuming that's a result of the old team's poor development practices.

---

Normally we can forego a `catch` statement and let our code explode unless we explicitly want to handle an error a certain way (with some sort of UI display normally). That way we, and our clients, can be sure that something is going wrong instead of guessing.

---

Ah, good to know. That's a fair point. I shall remove the empty catch statements! There shouldn't be any hidden issues there anyways.